src/org/scratchOrgSettingsGenerator.ts

```diff
@@ -209,9 +233,13 @@ export default class SettingsGenerator {
     allbusinessProcesses: string[]
   ) {
     if (this.objectSettingsData) {
-      for (const item of Object.keys(this.objectSettingsData)) {
-        const value: ObjectSetting = this.objectSettingsData[item];
-        const fileContent = this.createObjectFileContent(upperFirst(item), value, allRecordTypes, allbusinessProcesses);
+      for (const [item, value] of Object.entries(this.objectSettingsData)) {
+        const fileContent = this.createRecordTypeAndBusinessProcessFileContent(
+          upperFirst(item),
```

Comment:
pass in the item, and let upperFirst happen in the function.

you can make it const inside there if you need upperFirst in several places